### PR TITLE
Ter 103 external bucket drop role

### PR DESCRIPTION
**Pull request by martinjunger**
Jira: TER-103
Connection: https://github.com/keboola/connection/pull/4193/files

---

Before asking for review make sure that:

## Checklist

- [x] New client method(s) has tests
- [x] Apiary file is updated

## Release

  - [x] I gave the PR a proper label:
    * major (BC break)
    * minor (new feature)
    * patch (backwards compatible fix)
    * no release (just test changes)

### CODE REVIEW

**Comment by martinjunger** [comment]
**Snowflake** funguje ✅ 

**Teradata** by mela taky, ale bohuzel se na lokale nedostanu pres vytvoreni [WorkspaceBackendu](https://github.com/keboola/storage-api-php-client/pull/997/files#diff-85f40dda8f16faf9c11f37d4ca0bf2950c2cbf7defc30d2ae9c024abe210313cR246) (line 246) ⚠️ , pise to:
```
Doctrine\DBAL\Exception\DriverException : An exception occurred in the driver: odbc_connect(): SQL error: [Teradata][ODBC Teradata Driver] (130) TLS connection failed with SSL_ERROR_SYSCALL error:00000000:lib(0):func(0):reason(0), SQL state S1000 in SQLConnect
```
Kdyz si vypisu pripojovaci udaje a pripojim se v PhpStormu, tak to jde 🤷 .

👉 Potreboval bych, aby to jeste zkusil nekdo jiny...

File: tests/Backend/ExternalBuckets/TeradataRegisterBucketTest.php [hunk 1]
```diff
@@ -260,6 +260,29 @@ public function testRegisterExternalDB(): void
...
+            // TODO check correct message
+            $this->assertMatchesRegularExpression(
+                "/Database 'TEST_EXTERNAL_BUCKETS' does not exist or not authorized/",
+                $e->getMessage(),
+            );
+        } catch (\Throwable $e) {
+            // TODO check correct message
+            dump(get_class($e));
+            dump($e->getMessage());
+        }
```
Comment (martinjunger) [inline tests/Backend/ExternalBuckets/TeradataRegisterBucketTest.php:286]: TODO nahradit za spravnou message

**Comment by martinjunger** [comment]
> **Teradata** by mela taky, ale bohuzel se na lokale nedostanu pres vytvoreni [WorkspaceBackendu](https://github.com/keboola/storage-api-php-client/pull/997/files#diff-85f40dda8f16faf9c11f37d4ca0bf2950c2cbf7defc30d2ae9c024abe210313cR246) (line 246) ⚠️ , pise to:
> ...
> 👉 Potreboval bych, aby to jeste zkusil nekdo jiny...

Pustil jsem [tag](https://dev.azure.com/keboola-dev/Connection%20build/_build/results?buildId=30633&view=results), tak se to snad dozvim...

**Comment by martinjunger** [comment]
> > **Teradata** by mela taky, ale bohuzel se na lokale nedostanu pres vytvoreni [WorkspaceBackendu](https://github.com/keboola/storage-api-php-client/pull/997/files#diff-85f40dda8f16faf9c11f37d4ca0bf2950c2cbf7defc30d2ae9c024abe210313cR246) (line 246) ⚠️ , pise to:
> > ...
> > 👉 Potreboval bych, aby to jeste zkusil nekdo jiny...
> 
> Pustil jsem [tag](https://dev.azure.com/keboola-dev/Connection%20build/_build/results?buildId=30633&view=results), tak se to snad dozvim...

Tak nedozvim, `paratest` nic nevypisuje... :-/

File: tests/Backend/ExternalBuckets/TeradataRegisterBucketTest.php (hunk 1 above)
Comment (martinjunger) [reply tests/Backend/ExternalBuckets/TeradataRegisterBucketTest.php:286]: Upravil jsem na `The user does not have SELECT access to`, ktera by mela prijit...

**Comment by jirkasemmler** [comment]
@martinjunger fixnul jsem ti tu error message https://github.com/keboola/storage-api-php-client/pull/997/commits/d774bbc0529d9566ab6e19acdbefa56bb4aebd7d

**Review by jirkasemmler — APPROVED** [review]

**Comment by martinjunger** [comment]
ℹ️ rebase -> ready to merge

**Pull request merged**